RELEASING.md

```diff
@@ -1,36 +1,41 @@
-# Making a rustls release
+## Before making a release
 
-This is a checklist for steps to make before/after making a rustls release.
-
-1. Attend to the README.md: this appears on crates.io for the release, and can't be edited after
-   the fact.
-   - Ensure the version has a good set of release notes.  Move old release notes to OLDCHANGES.md
-     if this is getting excessively long.
-   - Write the version and date of the release.
-2. Run `cargo update` followed by `cargo outdated`, to check if we have any
+1. Run `cargo update` followed by `cargo outdated`, to check if we have any
    dependency updates which are not already automatically taken by their semver specs.
    - If we do, take them if possible with separate commits (but there should've been
      dependabot PRs submitted for these already.)
-3. Now run `cargo test --all-features` to ensure our tests continue to pass with the
+2. Now run `cargo test --all-features` to ensure our tests continue to pass with the
    updated dependencies.
-4. Update `rustls/Cargo.toml` to set the correct version.
-5. Make a commit with the above changes, something like 'Prepare $VERSION'.  This
+3. Update `rustls/Cargo.toml` to set the correct version.
+4. Make a commit with the above changes, something like 'Prepare $VERSION'.  This
    should not contain functional changes: just versions numbers, and markdown changes.
-6. Do a dry run: in `rustls/` check `cargo publish --dry-run`
-7. Push the above commit.  Wait for CI to confirm it as green.
+5. Do a dry run: in `rustls/` check `cargo publish --dry-run`
+6. Open a PR with the above commit.  Wait for review and CI to confirm it as green.
    - Any red _should_ naturally block the release.
    - If rustc nightly is broken, this _may_ be acceptable if the reason is understood
      and does not point to a defect in rustls.  eg, at the time of writing in releasing 0.20:
      - `cargo fuzz` is broken: https://github.com/rust-fuzz/cargo-fuzz/issues/276
      - oss fuzz is broken: https://github.com/google/oss-fuzz/issues/6268
      (Both of these share the same root cause of LLVM13 breaking changes; which are
       unfortunately common when rustc nightly takes a new LLVM.)
-8. Tag the released version: `git tag -m '0.20.0' v/0.20.0`
-9. Push the tag: `git push --tags`
-10. Do the release: `cargo publish` when sat in `rustls/`.
 
-## Post-release things
+## Making a release
 
-- Update dependent crates (eg, hyper-rustls, rustls-native-certs, etc.)
+1. Tag the released version: eg. `git tag -m '0.20.0' v/0.20.0`
+2. Push the tag: eg. `git push origin v/0.20.0`
+3. Do the release: `cargo publish` when sat in `rustls/`.
 
+## After making a release
 
+1. Create a new GitHub release for that tag.  Use "Generate release notes" (against the tag for the previous release)
```

Comment:
Huh, I had missed this button before.

The part I don't like about this process is that there is new review of the release notes. I feel like the release notes should be reviewed, ideally as part of the release prep PR.

Reply:
mmm i agree that is not ideal. will move that step to earlier so they can be reviewed in the PR

Reply:
just about to squash everything together so this thread might disappear, but i've moved the hand-written "headlines" part into the release PR, and then the release notes are this (post-review) plus the auto-generated parts.